lib/msal-node/src/config/Configuration.ts

```diff
@@ -107,6 +108,17 @@ export type Configuration = {
     telemetry?: NodeTelemetryOptions;
 };
 
+export type ManagedIdentityIdParams = {
+    userAssignedClientId?: string;
+    userAssignedResourceId?: string;
+    userAssignedObjectId?: string;
+};
+
+export type ManagedIdentityConfiguration = {
+    managedIdentityIdParams?: ManagedIdentityIdParams;
+    system?: NodeSystemOptions;
+};
```

Comment:
What about instead of `ManagedIdentityIdParams`, `ManagedIdentityConfiguration` accepts `ManagedIdentityIdType` (required) and `UserAssignedId` (optional, in case of system assigned Id) to make API clearer. Currently with `ManagedIdentityIdParams`, what if multiple IDs are set, what if none are set?

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/AppConfig/ManagedIdentityId.cs